_tips/developing.ja.md

```diff
@@ -0,0 +1,642 @@
+ref: /tips/developing/  # Do not change this
+
+ext_css: tips.css
+title_icon: /tips/img/icons.svg#developing
+
+acknowledgements: /tips/acknowledgements/
+
+# In the footer below:
+# Do not change the dates
+# Do not translate ACKNOWLEDGEMENTS
+# Translate the other words, including "Date:" and "Editors:"
+# Translate the Working Group name. Leave the Working Group acronym in English.
+footer: >
+  <p><strong>日にち:</strong> マイナーアップデート:2019年1月9日､アップデート:2016年4月15日｡最初の出版:2015年9月</p>
+  <p><strong>編集者:</strong> <a href="https://www.w3.org/People/kevin">Kevin White</a>, <a href="https://www.w3.org/People/shadi">Shadi Abou-Zahra</a>, and <a href="https://www.w3.org/People/Shawn">Shawn Lawton Henry</a>. ACKNOWLEDGEMENTS lists contributors and credits.</p>
+  <p><a href="https://www.w3.org/WAI/EO/">教育と援助のためのWG (EOWG)</a>によって開発され. <a href="https://www.w3.org/WAI/DEV/">WAI-DEV プロジェクト</a>からのサポートを受けるとともに, ヨーロッパ委員会の援助を受けた｡</p>
+---
+
+{::nomarkdown}
+{% include box.html type="start" h="2" title="概要" class="full" %}
+{:/}
+
+このページでは､障害者にとってアクセシブルなWebコンテンツの開発を始める際に役立つ､いくつかの基本的な考慮事項を紹介します｡これらのヒントは､Webコンテンツ・アクセシビリティ・ガイドライン(WCAG)の要件を満たすためのグッドプラクティスです｡関連するWCAG要件､「より理解するための」ドキュメントでの詳細な背景､チュートリアルからのガイダンス､ユーザーストーリーなどへのリンクをたどってください｡
+
+{::nomarkdown}
+{% include box.html type="end" %}
+{:/}
+
+{::options toc_levels="2" /}
+
+{::nomarkdown}
+{% include_cached toc.html type="start" title="Page Contents" class="full" %}
+{:/}
+
+-   TOC is created automatically.
+{:toc}
+
+{::nomarkdown}
+{% include_cached toc.html type="end" %}
+{:/}
+
+## すべてのフォームコントロールにラベルを関連付ける
+
+フォーム要素の `id` 属性にリンクされた `<label>` 要素の `for` 性を使うか､ <abbr title="Accessible Rich Internet Applications">WAI-ARIA</abbr> 属性を使います｡特定の状況では `<label>` 要素を視覚的に隠すことが許容されるかもしれませんが､ほとんどの場合ラベルは全ての読者が必要な入力を理解するのに必要です｡
+
+{::nomarkdown}
+{% include_cached box.html type="start" title="例:  <code>for</code>　属性と <code>id</code> 属性の使用" class="example" %}
+
+<div class="two-column">
+<figure>
+<figcaption>{% include_cached icon.html name="desktop" %} 表現</figcaption>
+<label for="username">ユーザーネーム</label>
+<input id="username" type="text" name="ユーザーネーム">
+</figure>
+<figure>
+<figcaption>{% include_cached icon.html name="code" %} 短いコード</figcaption>
+{:/}
+
+~~~html
+<label for="username">ユーザーネーム</label>
+<input id="username" type="text" name="ユーザーネーム">
+~~~
+
+{::nomarkdown}
+</figure>
+</div>
+{% include_cached box.html type="end" %}
+{:/}
+
+{::nomarkdown}
+{% include_cached box.html type="start" title="詳細情報" class="simple" %}
+{:/}
+
+* **WCAG**
+  * [ラベルあるいは説明 3.3.2](/WAI/WCAG21/quickref/#labels-or-instructions) ([理解のため 3.3.2](/WAI/WCAG21/Understanding/labels-or-instructions))
+* **Tutorial**
+  * [ラベルの制御](/tutorials/forms/labels/)
+
+{::nomarkdown}
+{% include_cached box.html type="end" %}
+{:/}
+
+## 画像の代替テキストを含める
+
+すべての情報画像と機能画像に画像の代替テキストが追加されていることを確認すること｡装飾的な画像には空の代替テキスト､`alt=""`を使うか､代わりにCSSに含めてください｡代替テキストは通常､文章コンテンツの責任者が提供します｡
```

Comment:
@jaykishigami Can you confirm to me the translation is equivalent to "Use empty alternative text, alt="" for decorative images, or include _[the decorative images]_ in the CSS instead"

I want to make sure the translation does not imply we recommend including the alternative text in the CSS.